password_enhancements.module

```diff
@@ -522,8 +553,277 @@ function password_enhancements_legal_allowed_paths_alter(array &$allowed_paths):
 }
 
 /**
- * Implements hook_config_translation_info_alter().
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_user_role_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Bail out early without the "user password enhancements settings" permission.
+  if (!\Drupal::currentUser()->hasPermission('user password enhancements settings')) {
+    return;
+  }
+  /** @var \Drupal\user\RoleForm $form_object */
+  $form_object = $form_state->getFormObject();
+  /** @var \Drupal\user\Entity\Role $role */
+  $role = $form_object->getEntity();
+  /** @var \Drupal\password_enhancements\PasswordPolicy $policy */
+  $policy = PasswordPolicy::createFromRole($role);
+  if ($role->id() != AccountInterface::ANONYMOUS_ROLE) {
+    // A master killswitch to allow removing the policy from the role.
+    $form['password_enhancements_apply_policy'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Apply password policy to this role'),
+      '#default_value' => !empty($role->getThirdPartySettings('password_enhancements')),
+    ];
+
+    $form['password_enhancements'] = [
+      '#tree' => TRUE,
+      '#type' => 'fieldset',
+      '#title' => t('Password policy'),
+      '#states' => [
+        'visible' => [
+          ':input[name="password_enhancements_apply_policy"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['password_enhancements']['minimumRequiredConstraints'] = [
+      '#type' => 'number',
+      '#title' => t('Minimum required constraints'),
+      '#min' => 1,
+      '#required' => TRUE,
+      '#default_value' => $policy ? $policy->getMinimumRequiredConstraints() : 1,
+    ];
+
+    $expire_days = $policy ? $policy->getExpireDays() : PasswordPolicyInterface::PASSWORD_NO_EXPIRY;
+    $form['password_enhancements']['expire_password'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Expire password'),
+      '#default_value' => $expire_days > 0,
+    ];
+
+    $form['password_enhancements']['expire_days'] = [
+      '#type' => 'number',
+      '#title' => t('Expire password after'),
+      '#field_suffix' => t('day(s).'),
+      '#default_value' => $expire_days ?: 1,
+      '#min' => 1,
+      '#states' => [
+        'visible' => [
+          ':input[name="password_enhancements[expire_password]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['password_enhancements']['expire_warn_before_days'] = [
+      '#type' => 'number',
+      '#title' => t('Show warning'),
+      '#description' => t('Show a warning message before the password would expire.<br>If set to 0 it will not show any message.'),
+      '#field_suffix' => t('day(s) before the password expires.'),
```

Comment:
It looks odd to me and also rings a bell from last week: https://www.drupal.org/project/drupal/issues/3072752#comment-13213805

Reply:
As this `t()` spits out only text without HTML, I think we're fine.